src/Bicep.Core/Workspaces/SourceFileGroupingBuilder.cs

```diff
@@ -25,7 +25,7 @@ public class SourceFileGroupingBuilder
         private readonly IReadOnlyWorkspace workspace;
 
         private readonly Dictionary<Uri, ResultWithDiagnostic<ISourceFile>> fileResultByUri;
-        private readonly ConcurrentDictionary<ISourceFile, Dictionary<IArtifactReferenceSyntax, Result<Uri, UriResolutionError>>> uriResultByArtifactReference;
+        private readonly ConcurrentDictionary<ISourceFile, Dictionary<IArtifactReferenceSyntax, Result<Uri, UriResolutionError>>> fileUriResultByArtifactReference;
```

Comment:
That is a fair comment, I am trying hard to keep things separate. The issue here is that I feel this field was a bit too vague. It wasn't clear the URIs are only URIs in the local filesystem. But the feedback is noted and I will be taking into consideration in all further PRs :)

Reply:
Agreed, although I understand sometimes the extra time on your part is not worth it.  This one hasn't been too bad IMHO in spite of the large number of changes because most of the refactorings are straightforward and don't cause file renames.